workflow/common/util.go

```diff
@@ -263,6 +263,18 @@ func ProcessArgs(tmpl *wfv1.Template, args wfv1.ArgumentsProvider, globalParams,
 		if argParam != nil && argParam.Value != nil {
 			inParam.Value = argParam.Value
 		}
+
+		if inParam.ValueFrom != nil && inParam.ValueFrom.ConfigMapKeyRef != nil {
+			inParam.Value = wfv1.AnyStringPtr("Generated at runtime using config maps")
+			if clientset != nil {
+				configMapValue, err := util.GetConfigMaps(*clientset, namespace, inParam.ValueFrom.ConfigMapKeyRef.Name, inParam.ValueFrom.ConfigMapKeyRef.Key)
```

Comment:
I would turn the GetConfigMaps code into an isntanciated service that has an instance of clientset instead of passing it here

Reply:
Any example of such a service in the source code would be helpful here to piggy-back off of

Reply:
Does it make sense to add something similar to the `cache` package for k8s resources like `configmaps` and `secrets` and add it to the controller instance itself?

Reply:
Sorry for the delay in answering. Take a look at https://github.com/argoproj/argo/tree/master/workflow/controller/cache and how it's used. This provides a good example of what I mean